burn-tensor/src/tensor/container.rs

```diff
@@ -1,17 +1,23 @@
 use crate::{backend::Backend, Tensor};
-use std::{any::Any, collections::HashMap};
+use core::any::Any;
+use core::marker::PhantomData;
+
+extern crate alloc;
+use alloc::boxed::Box;
+
+use alloc::collections::BTreeMap;
 
 /// Contains tensor of arbitrary dimension.
 #[derive(Debug)]
 pub struct TensorContainer<B: Backend, ID> {
-    tensors: HashMap<ID, Box<dyn Any + Send + Sync>>,
-    _b: B,
+    tensors: BTreeMap<ID, Box<dyn Any + Send + Sync>>,
```

Comment:
Hmm, ok so I understand now why you needed `Ord` and `PartialOrd` for the ID struct. `HashMap` might still be more performant, so we may switch to a `HashMap` when `no_std` is not activated.

Reply:
OK. I'll add behind the feature flag.